src/common.h

```diff
@@ -174,6 +174,28 @@ GIT_INLINE(void) git__init_structure(void *structure, size_t len, unsigned int v
 	GITERR_CHECK_VERSION(&(VERSION), _tmpl.version, #TYPE);	\
 	memcpy((PTR), &_tmpl, sizeof(_tmpl)); } while (0)
 
+/** Check for integer overflow from addition or multiplication */
+#define GIT_ALLOC_OVERFLOW_ADD(one, two) \
+	((one) + (two) < (one))
```

Comment:
IIRC, this is undefined in C and might not work as intended. See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=30475 and https://gcc.gnu.org/bugzilla/show_bug.cgi?id=30475#c8